src/dune_rpc_impl/server.ml

```diff
@@ -329,25 +335,33 @@ let handler (t : _ t Fdecl.t) handle : 'build_arg Dune_rpc_server.Handler.t =
     Handler.declare_notification rpc Procedures.Server_side.log
   in
   let () = Handler.implement_request rpc Procedures.Public.ping (fun _ -> Fiber.return) in
-  let () =
-    let build _session targets =
+  let implement_request_pending_action decl ~f =
+    let handler _session input =
       let server = Fdecl.get t in
-      let ivar = Fiber.Ivar.create () in
-      let targets = List.map targets ~f:server.parse_build_arg in
-      let* () = Job_queue.write server.pending_build_jobs (targets, ivar) in
-      let+ build_outcome = Fiber.Ivar.read ivar in
+      let outcome = Fiber.Ivar.create () in
+      let* () = Job_queue.write server.pending_jobs { kind = f input; outcome } in
+      let+ build_outcome = Fiber.Ivar.read outcome in
       match (build_outcome : Build_outcome.t) with
       | Success -> Build_outcome_with_diagnostics.Success
       | Failure -> Failure (get_current_diagnostic_errors ())
     in
-    Handler.implement_request rpc Decl.build build
+    Handler.implement_request rpc decl handler
+  in
+  let () =
```

Comment:
Maybe just use `;`, to show that it is side-effectful and not expected to return any value.